lib/core/src/Cardano/Pool/Metadata.hs

```diff
@@ -116,6 +122,10 @@ metadaFetchEp pid (StakePoolMetadataHash bytes)
     hashStr = T.unpack $ T.decodeUtf8 $ convertToBase Base16 bytes
     pidStr  = T.unpack $ toText pid
 
+-- TODO: use SMASH servant types
```

Comment:
I disagree. The entire point of servant is to be able to consume its types.

> It may happen that other SMASH servers are used instead.

That sounds too theoretical to base implementation decisions on it.

Reply:
> Likewise, in README.md, I see a table containing SMASH software versions. That is not especially relevant to users, because they are not the ones operating the SMASH server(s).

Feel free to discuss that with @KtorZ who requested this change.

Reply:
> That sounds too theoretical to base implementation decisions on it.

It is already happening as we speak. Adapools has its own SMASH server. This is why I've also repeated in the past that we want to avoid re-using SMASH's internal types. We do integrate against the interface (a.k.a OpenAPI), not the implementation. 
Now, we'll probably end up using servant types _anyway_ on our end, so it could be interesting to define them in a library that is self contained, but this should not create any strong coupling with SMASH and SMASH development. 

> Likewise, in README.md, I see a table containing SMASH software versions. That is not especially relevant to users, because they are not the ones operating the SMASH server(s).

@rvl I've asked Julian to add that because it does actually concern users. Users are going to point cardano-wallet to SMASH servers and they want to know with which version of SMASH it works. And by version here, I mean the version of the corresponding OpenAPI specification, not the version of SMASH-the-haskell-executable (although arguably, major and minor should move in locksteps for these).

Reply:
> Now, we'll probably end up using servant types anyway on our end, so it could be interesting to define them in a library that is self contained, but this should not create any strong coupling with SMASH and SMASH development.

SMASH already exposes the servant types separately: https://github.com/input-output-hk/smash/tree/master/smash-servant-types